cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -840,21 +858,17 @@ producerAddressesNonP2P nt =
 
 producerAddresses
   :: NetworkTopology
-  -> ([(HotValency, WarmValency, Map RelayAccessPoint PeerAdvertise)], Map RelayAccessPoint PeerAdvertise)
-producerAddresses nt =
-  case nt of
-    RealNodeTopology lrpg prp _ ->
-      ( map (\lrp -> ( hotValency lrp
-                     , warmValency lrp
-                     , Map.fromList $ rootConfigToRelayAccessPoint
-                                    $ localRoots lrp
-                     )
-            )
-            (groups lrpg)
-      , foldMap (Map.fromList . rootConfigToRelayAccessPoint . publicRoots) prp
-      )
-
-useLedgerAfterSlot
-  :: NetworkTopology
-  -> UseLedgerAfter
-useLedgerAfterSlot (RealNodeTopology _ _ (UseLedger ul)) = ul
+  -> ([(HotValency, WarmValency, Map RelayAccessPoint (PeerAdvertise, PeerTrustable))], Map RelayAccessPoint PeerAdvertise)
+producerAddresses RealNodeTopology { ntLocalRootPeersGroups
+                                   , ntPublicRootPeers
+                                   } =
+  ( map (\lrp -> ( hotValency lrp
+                 , warmValency lrp
+                 , Map.fromList $ map (fmap (, trustable lrp))
```

Comment:
This is a bit ugly that we need to add this information this way. We cannot simply extend `RootConfig` because that type is used in various places.  A nicer way would be to provide `LocalRootConfig` type:
```hs
data LocalRootConfig = LocalRootConfig
  { localRootAccessPoints :: [RelayAccessPoint]
    -- ^ a list of relay access points, each of which is either an ip address
    -- or domain name and a port number.
  , localRootAdvertise    :: PeerAdvertise,
    -- ^ 'advertise' configures whether the root should be advertised through
    -- gossip.
  , localRootTrustable    :: Bool
    -- ^ ...
  } deriving (Eq, Show)

```